build-zfs-module/Containerfile

```diff
@@ -0,0 +1,30 @@
+FROM quay.io/fedora/fedora-coreos:stable as kernel-query
+#We can't use the `uname -r` as it will pick up the host kernel version
+RUN rpm -qa kernel --queryformat '%{VERSION}-%{RELEASE}.%{ARCH}' > /kernel-version.txt
+
+# Using https://openzfs.github.io/openzfs-docs/Developer%20Resources/Custom%20Packages.html
+FROM registry.fedoraproject.org/fedora:36 as builder
+COPY --from=kernel-query /kernel-version.txt /kernel-version.txt
+WORKDIR /etc/yum.repos.d
+RUN curl -L -O https://src.fedoraproject.org/rpms/fedora-repos/raw/f37/f/fedora-updates-archive.repo && \
+    sed -i 's/enabled=AUTO_VALUE/enabled=true/' fedora-updates-archive.repo
+RUN dnf install -y dkms gcc make autoconf automake libtool rpm-build libtirpc-devel libblkid-devel \
+    libuuid-devel libudev-devel openssl-devel zlib-devel libaio-devel libattr-devel elfutils-libelf-devel \
+    kernel-$(cat /kernel-version.txt) kernel-modules-$(cat /kernel-version.txt) kernel-devel-$(cat /kernel-version.txt) \
+    python3 python3-devel python3-setuptools python3-cffi libffi-devel git ncompress libcurl-devel
+WORKDIR /
+RUN curl -L -O https://github.com/openzfs/zfs/releases/download/zfs-2.1.6/zfs-2.1.6.tar.gz && tar xzf zfs-2.1.6.tar.gz
+WORKDIR /zfs-2.1.6
```

Comment:
A future improvement here would be to use a script or something to derive the latest release and not hardcode this version.